### PR TITLE
feat: client changes for query stats 

### PR DESCRIPTION
**Pull request by vikrantgupta25**
### Summary

this PR adds the query statistics for the logs explorer page 

#### Related Issues / PR's

fixes https://github.com/SigNoz/engineering-pod/issues/1661

#### Screenshots

https://github.com/user-attachments/assets/77b86769-77cd-4dde-94e9-12b18150acd8

### CODE REVIEW

**Comment by vikrantgupta25** [comment]
Do not merge this, raised against the feature branch

File: frontend/src/api/common/getQueryStats.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	queryId: string;
+	setData: React.Dispatch<React.SetStateAction<WsDataEvent | undefined>>;
+}
+
+export function getQueryStats(props: GetQueryStatsProps): void {
+	const { queryId, setData } = props;
+
+	const token = getLocalStorageApi(LOCALSTORAGE.AUTH_TOKEN) || '';
+	const socket = new WebSocket(
+		`ws://localhost:8080/api/v3/query_progress?q=${queryId}`,
```
Comment (vikrantgupta25) [inline frontend/src/api/common/getQueryStats.ts:19]: ignore this for now, will change this

File: frontend/src/api/common/getQueryStats.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	);
+
+	socket.addEventListener('message', (event) => {
+		try {
+			const parsedData = JSON.parse(event.data);
+			setData(parsedData);
+		} catch {
+			setData(event.data);
+		}
+	});
```
Comment (SagarRajput-7) [inline frontend/src/api/common/getQueryStats.ts:31]: any safety check needed here? specially for `event: Event` type, that imo dont by default have data prop

File: frontend/src/container/LogsExplorerViews/QueryStatus.styles.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.query-status {
+	display: flex;
+	align-items: center;
+	.loading {
+	}
+	.error {
+	}
+	.success {
+	}
+}
```
Comment (SagarRajput-7) [inline frontend/src/container/LogsExplorerViews/QueryStatus.styles.scss:10]: empty styles

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/container/LogsExplorerViews/QueryStatus.styles.scss (hunk 3 above)
Comment (vikrantgupta25) [reply frontend/src/container/LogsExplorerViews/QueryStatus.styles.scss:4]: removed

File: frontend/src/api/common/getQueryStats.ts (hunk 2 above)
Comment (vikrantgupta25) [reply frontend/src/api/common/getQueryStats.ts:31]: added

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/container/LogsExplorerViews/index.tsx [hunk 4]
```diff
@@ -116,6 +120,10 @@ function LogsExplorerViews({
 	const [logs, setLogs] = useState<ILog[]>([]);
 	const [requestData, setRequestData] = useState<Query | null>(null);
 	const [showFormatMenuItems, setShowFormatMenuItems] = useState(false);
+	const [queryId, setQueryId] = useState<string>(
+		Math.random().toString(36).slice(2),
```
Comment (YounixM) [inline frontend/src/container/LogsExplorerViews/index.tsx:124]: Can we not use a fixed constant here ?

File: frontend/src/container/LogsExplorerViews/index.tsx [hunk 5]
```diff
@@ -318,6 +336,23 @@ function LogsExplorerViews({
 		],
 	);
 
+	useEffect(() => {
+		setQueryId(Math.random().toString(36).slice(2));
```
Comment (YounixM) [inline frontend/src/container/LogsExplorerViews/index.tsx:340]: use uuid library

File: frontend/src/container/LogsExplorerViews/index.tsx [hunk 6]
```diff
@@ -318,6 +336,23 @@ function LogsExplorerViews({
...
+	useEffect(() => {
+		if (
+			!isEmpty(queryId) &&
+			(isLoading || isFetching) &&
+			selectedPanelType !== PANEL_TYPES.LIST
+		) {
+			setQueryStats(undefined);
+			setTimeout(() => {
+				getQueryStats({ queryId, setData: setQueryStats });
+			}, 500);
```
Comment (YounixM) [inline frontend/src/container/LogsExplorerViews/index.tsx:352]: Reason for delay ?

File: frontend/src/container/LogsExplorerViews/index.tsx [hunk 7]
```diff
@@ -665,6 +700,30 @@ function LogsExplorerViews({
...
+						{(selectedPanelType === PANEL_TYPES.TIME_SERIES ||
+							selectedPanelType === PANEL_TYPES.TABLE) && (
+							<div className="query-stats">
+								<QueryStatus
+									loading={isLoading || isFetching}
+									error={isError}
+									success={isSuccess}
+								/>
+								{queryStats?.read_rows && (
+									<Typography.Text className="rows">
```
Comment (YounixM) [inline frontend/src/container/LogsExplorerViews/index.tsx:712]: Please make sure fallbacks are in place here. Bad data shouldn't break the UI

File: frontend/src/container/LogsExplorerViews/index.tsx (hunk 4 above)
Comment (vikrantgupta25) [reply frontend/src/container/LogsExplorerViews/index.tsx:124]: no we cannot use a fixed constant as multiple users opening the UI at the same time will have the same id and the socket will get attached to wrong queries if we have the same initial ID for everyone

File: frontend/src/container/LogsExplorerViews/index.tsx (hunk 6 above)
Comment (vikrantgupta25) [reply frontend/src/container/LogsExplorerViews/index.tsx:352]: there is a delay between the `query_range` API reaching the query service and ws connection, if the connection is being tried to establish before the query_range API reaching the qs it fails hence the delay.

File: frontend/src/container/LogsExplorerViews/index.tsx (hunk 7 above)
Comment (vikrantgupta25) [reply frontend/src/container/LogsExplorerViews/index.tsx:712]: if we do not receive any data we do not show anything in the UI, just the status of the request.

File: frontend/src/container/LogsExplorerViews/index.tsx (hunk 5 above)
Comment (vikrantgupta25) [reply frontend/src/container/LogsExplorerViews/index.tsx]: done

**Review by YounixM — DISMISSED** [review]
LGTM. 

Please get review from a BE team member for API related changes.

**Comment by vikrantgupta25** [comment]
bad rebase! reopened here - https://github.com/SigNoz/signoz/pull/5706

**Pull request closed without merging**